src/browser/modules/Sidebar/project-files.utils.ts

```diff
@@ -119,6 +122,55 @@ export const createFilePath = (paths: string[]): string => {
   return paths.join(/Win32/.test(navigator.platform) ? '\\' : '/')
 }
 
+export const checkFileNameInput = (fileName: string): string => {
+  if (!fileName.length) {
+    return 'File name cannot be empty'
+  }
+
+  const errors: ProhibitedFilenameErrors = { chars: [], tests: [] }
+
+  PROHIBITED_FILENAME_CHAR_TESTS.forEach(charObj => {
+    if (charObj.test) {
+      // if a Regex test exists
+      if (charObj.platform) {
+        if (
+          new RegExp(charObj.platform).test(navigator.platform) &&
+          charObj.test.test(fileName)
+        ) {
+          errors.tests.push(charObj.test)
+        }
+      } else {
+        if (charObj.test.test(fileName)) {
+          errors.tests.push(charObj.test)
+        }
+      }
+    }
+    if (charObj.char) {
+      // otherwise default to .includes check
+      if (charObj.platform) {
+        if (
+          new RegExp(charObj.platform).test(navigator.platform) &&
+          fileName.includes(charObj.char)
+        ) {
+          errors.chars.push(charObj.char)
+        }
+      } else {
+        if (fileName.includes(charObj.char)) {
+          errors.chars.push(charObj.char)
+        }
+      }
+    }
+  })
+
+  return errors.chars.length || errors.tests.length
+    ? `File name cannot contain ${
+        errors.chars.length ? `${errors.chars.join(', ')}` : ''
+      }${errors.chars.length && errors.tests.length ? ' or ' : ''}${
+        errors.tests.length ? `${errors.tests.join(', ')}` : ''
```

Comment:
was going to present regex tests in a different way but couldnt think of anything for now, so left it like this